src/models/tag.js

```diff
@@ -53,5 +53,31 @@ export default {
       (item) => { items.push(item.id) })
 
     return items
+  },
+
+  async merge(db, ids) {
+    let [keepId, ...mergeIds] = ids
+    let itemsWithOldTags = new Set()
+    // find all item ids which have taggings of tags to be deleted
+    for (const id of mergeIds) {
+      itemsWithOldTags.add(...await this.items(db, id))
+    }
+
+    // delete all tags except the one to be kept
+    // on delete cascade causes taggings to be removed too
+    await this.delete(db, mergeIds)
+
+    // recreate taggings for items which had deleted tags
+    for (const item of itemsWithOldTags) {
```

Comment:
Lets fold the loop into the DB command. This way we should be able to send a single insert command for all items to the DB instead of one command for each item.